src/probability/kernel/basic.lean

```diff
@@ -289,6 +293,239 @@ end
+        have habj : {(a, b)} ⊆ f j, by { rw set.singleton_subset_iff, exact hsj hbs, },
+        simpa only [set.bot_eq_empty, set.le_eq_subset, set.singleton_subset_iff,
+          set.mem_empty_iff_false] using h_disj hij habi habj, },
+      { exact λ i, (@measurable_prod_mk_left α β _ _ a) _ (hf_meas i), }, },
+    rw h_tsum,
+    exact measurable.ennreal_tsum hf, },
+end
+
+lemma measurable_prod_mk_mem (κ : kernel α β) [is_s_finite_kernel κ]
+  {t : set (α × β)} (ht : measurable_set t) :
+  measurable (λ a, κ a {b | (a, b) ∈ t}) :=
+begin
+  rw ← kernel_sum_seq κ,
+  have : ∀ a, kernel.sum (seq κ) a {b : β | (a, b) ∈ t} = ∑' n, seq κ n a {b : β | (a, b) ∈ t},
+    from λ a, kernel.sum_apply' _ _ (measurable_prod_mk_left ht),
+  simp_rw this,
+  refine measurable.ennreal_tsum (λ n, _),
+  exact measurable_prod_mk_mem_of_finite (seq κ n) ht infer_instance,
+end
+
+lemma measurable_lintegral_indicator_const (κ : kernel α β) [is_s_finite_kernel κ]
+  {t : set (α × β)} (ht : measurable_set t) (c : ℝ≥0∞) :
+  measurable (λ a, ∫⁻ b, t.indicator (function.const (α × β) c) (a, b) ∂κ a) :=
+begin
+  simp_rw lintegral_indicator_const_comp measurable_prod_mk_left ht _,
+  exact measurable.const_mul (measurable_prod_mk_mem _ ht) c,
+end
+
+/-- For an s-finite kernel `κ` and a function `f : α → β → ℝ≥0∞` which is measurable when seen as a
+map from `α × β` (hypothesis `measurable (function.uncurry f)`), the integral
+`a ↦ ∫⁻ b, f a b ∂(κ a)` is measurable. -/
+theorem measurable_lintegral (κ : kernel α β) [is_s_finite_kernel κ]
+  (f : α → β → ℝ≥0∞) (hf : measurable (function.uncurry f)) :
+  measurable (λ a, ∫⁻ b, f a b ∂κ a) :=
+begin
+  let F : ℕ → simple_func (α × β) ℝ≥0∞ := simple_func.eapprox (function.uncurry f),
+  have h : ∀ a, (⨆ n, F n a) = function.uncurry f a,
+    from simple_func.supr_eapprox_apply (function.uncurry f) hf,
+  simp only [prod.forall, function.uncurry_apply_pair] at h,
+  simp_rw ← h,
+  have : ∀ a, ∫⁻ b, (⨆ n, F n (a, b)) ∂κ a = ⨆ n, ∫⁻ b, F n (a, b) ∂κ a,
+  { intro a,
+    rw lintegral_supr,
+    { exact λ n, (F n).measurable.comp measurable_prod_mk_left, },
+    { exact λ i j hij b, simple_func.monotone_eapprox (function.uncurry f) hij _, }, },
+  simp_rw this,
+  refine measurable_supr (λ n, simple_func.induction _ _ (F n)),
+  { intros c t ht,
+    simp only [simple_func.const_zero, simple_func.coe_piecewise, simple_func.coe_const,
+      simple_func.coe_zero, set.piecewise_eq_indicator],
+    exact measurable_lintegral_indicator_const κ ht c, },
+  { intros g₁ g₂ h_disj hm₁ hm₂,
+    simp only [simple_func.coe_add, pi.add_apply],
+    have h_add : (λ a, ∫⁻ b, g₁ (a, b) + g₂ (a, b) ∂κ a)
+      = (λ a, ∫⁻ b, g₁ (a, b) ∂κ a) + (λ a, ∫⁻ b, g₂ (a, b) ∂κ a),
+    { ext1 a,
+      rw [pi.add_apply, lintegral_add_left (g₁.measurable.comp measurable_prod_mk_left)], },
+    rw h_add,
+    exact measurable.add hm₁ hm₂, },
+end
+
+lemma measurable_set_lintegral (κ : kernel α β) [is_s_finite_kernel κ]
```

Comment:
Can you add the corollaries with `f : α→ ℝ≥0∞` as well?

Reply:
Would such a corollary be really useful though? In that case the integral simplifies to `f a * κ a s` (by simp), so the lemma would show measurability of an integral that would rarely appear in goals.
I guess that I can add the other corollary, with `f : β → ℝ≥0∞`, showing `measurable (λ a, ∫⁻ b in s, f b ∂κ a)`.